web/src/main/webapp/heatMapArray.ts

```diff
@@ -71,99 +83,125 @@ export class LegendOverlay implements IHtmlElement {
     private xText: any;
     private yText: any;
 
-    // Fields for color legend
-    private colorMap: ColorMap;
-    // g element that will contain the legend.
-    private legendG: any;
+    constructor(
+        private parent, // Element where this heat map is appended to.
```

Comment:
Sure, I've added the type, but since it's a d3 selection it is either the (quite ugly) `d3.Selection<Element, {}, null, undefined>`, which I don't fully understand and seems a bit excessive, or `any`. Do you know a way to to this cleaner?

Reply:
Use any and add a comment.
The types for d3 are not very good.